xdsl/dialects/pdl.py

```diff
@@ -249,15 +249,18 @@ class OperationOp(IRDLOperation):
     def __init__(
         self,
         op_name: str | StringAttr | None,
-        attribute_value_names: ArrayAttr[StringAttr] | None = None,
+        attribute_value_names: Iterable[StringAttr] | None = None,
```

Comment:
i think it gets us further away from being able to generate them automatically, and it just hides bits of complexity in lots of functions around the framework

Reply:
Ultimately, the goal in my mind is a frontend that makes it super conventient to construct these things the first time, and then rewrites that just pass the things around

Reply:
Well, then we need to make a decision on the rest of the codebase, because there are other places in the code where we use an `Iterable`, like for `shape` in some `builtin` and `memref`.

I'm not completely sure that we should restraint ourself because of what we want though. I'm fine removing some constructors if we realize that we can't autogenerate them, but I would prefer autogenerating them, and then building on top of them or something. Ultimately, we want to make our current code better, as right now we don't have a clear plan on how to generate everything?

I'll put the discussion on Zulip, that's probably better.

Reply:
No, you're right, in the short term this makes sense